### PR TITLE
Implement Mailing

### LINKED ISSUE #17: Add mail support
- [x] Send invitation links to mail
- [x] Add testmail functionality

## Sending opt-in monthly reports
- [x] Add account setting for opt-int to monthly reports
- [x] Add UI option to enable reports
- [x] implement mail sending
- [x] Assemble report
- [x] Restrict enabling monthly reports to accounts with mail address
- [x] disable monthly reports when removing mail address

## Finishing up
- [x] Run rustfmt on all edited files

### PR DESCRIPTION
**Pull request by Feliix42**
This PR closes #17 and adds mailing functionalities.

~Will remove the `[WIP]` label when it's ready for merge.~

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1 +1,7 @@
 DATABASE_URL=postgres://ascii:ascii@localhost:54320/ascii-pay
+
+# Submission Port is 587 per default.
+MAIL_SENDER=payments@ascii.coffee
+MAIL_URL=mail.example.local
+MAIL_USER=username
+MAIL_PASSWORD=password
```
Comment (V1ncNet) [inline .env]: Consider new line at the end of files

Comment (Feliix42) [reply .env]: Huh, this was unintentional. Surely a glitch in the editor config.

File: src/core/mail.rs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    }
+}
+
+fn send_standard_mail(account: &Account, subj: &str, message: String) -> ServiceResult<()> {
+    let credentials = MailCredentials::load_from_environment();
+
+    let email = EmailBuilder::new()
+        // Addresses can be specified by the tuple (email, alias)
+        .to((
+            account.mail.as_ref().expect("No mail address provided"),
```
Comment (pixix4) [inline src/core/mail.rs]: Please change `expect` to `ServiceError`

File: src/core/mail.rs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+pub fn send_test_mail(receiver: String) -> ServiceResult<()> {
+    let credentials = MailCredentials::load_from_environment();
+
+    let mail = EmailBuilder::new()
+        .to(receiver)
+        .from(credentials.sender)
+        .subject("[ascii pay] Test Mail")
+        .text("This is a test mail to verify that the miling system works.")
+        .build()
+        .unwrap();
```
Comment (pixix4) [inline src/core/mail.rs]: Please change `unwrap` to `ServiceError`

**Review by pixix4 — CHANGES_REQUESTED** [review]

**Review by pixix4 — APPROVED** [review]

**Pull request merged**